src/main/java/com/mraof/minestuck/entity/ai/MobAnimationPhaseGoal.java

```diff
@@ -0,0 +1,89 @@
+package com.mraof.minestuck.entity.ai;
+
+import com.mojang.math.Vector3d;
+import com.mraof.minestuck.entity.AnimatedPathfinderMob;
+import com.mraof.minestuck.entity.animation.PhasedMobAnimation;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.ai.goal.Goal;
+
+import java.util.EnumSet;
+import java.util.Objects;
+
+/**
+ * An abstract goal for handling PhasedMobAnimation compliant animated actions
+ */
+public abstract class MobAnimationPhaseGoal<T extends PathfinderMob & PhasedMobAnimation.Phases.Holder> extends Goal
+{
+	protected final T entity;
+	protected final PhasedMobAnimation animation;
+	
+	/**
+	 * Starts at 0 and counts up, when this int matches any of the ints in phases, it transitions to the start of that given phase
+	 */
+	protected int time = 0;
+	
+	protected Vector3d lookTarget;
+	
+	public MobAnimationPhaseGoal(T entity, PhasedMobAnimation animation)
+	{
+		this.entity = entity;
+		this.animation = animation;
+		
+		//code block has some redundancy with the tick function in AnimatedPathfinderMob, however entities who are not members of that class are expected to be able to use this goal
```

Comment:
This comment is no longer accurate. Ideally I hope that there would be no redundancy at all, but that'd require some replacements to the freezing/unfreezing in `AnimatedPathfinderMob` which I don't think is a must-fix for this PR.

Just update the comment for now.

Reply:
> with the tick function in AnimatedPathfinderMob